map.go

```diff
@@ -73,6 +74,17 @@ func (m *BPFMap) ReuseFD(fd int) error {
 	return nil
 }
 
+func (m *BPFMap) AttachStructOps() error {
+	if m.Type().String() != MapTypeStructOps.String() {
+		return errors.New("Map type should be BPF_MAP_TYPE_STRUCT_OPS")
+	}
+	linkC, errno := C.bpf_map__attach_struct_ops(m.bpfMap)
+	if linkC == nil {
+		return fmt.Errorf("Map attach failed: %v", &errno)
+	}
+	return nil
```

Comment:
We need to take care of this pointer as `bpf_link__detach_struct_ops()` receives it, so AttachStructOps() should return a *BPFLink.

P.S.: `bpf_link__detach_struct_ops()` wrapper should be implemented as well. Take a look at the BPFLink type - currently it holds and handles only BPFProg, but you might change it to accomodate a BPFMap as well, I believe.

Reply:
Hi @geyslan 
bpf_link__detach_struct_ops is not public API in the current, do you have any idea on how to handle this?
thanks!

Reply:
@geyslan 
If my understanding is correct, the `bpf_link__detach_struct_ops()` will be invoked when `link.Destroy()` called.
![image](https://github.com/user-attachments/assets/9f9d75df-8260-4c25-8b42-d68381631c37)
![image](https://github.com/user-attachments/assets/d507cd7d-f27d-40cc-9b56-a89170906b79)

So we don't need to add wrapped function for detaching strectOps map, right?